tests/utils.go

```diff
@@ -3364,16 +3364,20 @@ func CreateISCSITargetPOD(containerDiskName cd.ContainerDisk) (iscsiTargetIP str
 	}
 
 	pod, err = virtClient.CoreV1().Pods(NamespaceTestDefault).Create(pod)
-	PanicOnError(err)
+	Expect(err).ToNot(HaveOccurred(), "should successfully create ISCSI target pod")
 
 	getStatus := func() k8sv1.PodPhase {
 		pod, err := virtClient.CoreV1().Pods(NamespaceTestDefault).Get(pod.Name, metav1.GetOptions{})
-		Expect(err).ToNot(HaveOccurred())
-		iscsiTargetIP = pod.Status.PodIP
+		Expect(err).NotTo(HaveOccurred(), "should fetch ISCSI target pod for 'Status.Phase' checking")
 		return pod.Status.Phase
 	}
-	Eventually(getStatus, 120, 1).Should(Equal(k8sv1.PodRunning))
-	return
+	Eventually(getStatus, 120, 1).Should(Equal(k8sv1.PodRunning), "should fetch running ISCSI target pod")
+
+	// Fetch updated pod object after phase changed to Running to ensure Status.podIPs is not empty
+	pod, err = virtClient.CoreV1().Pods(NamespaceTestDefault).Get(pod.Name, metav1.GetOptions{})
+	Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("should sucessfully get pod %s after phase changed to Running", pod.Name))
```

Comment:
You could actually take that to a next step, and once the pod is ready, return its IP address. Just a thought.